switched.c

```diff
@@ -127,16 +172,26 @@ int main(int argc, char **argv) {
             return 2;
         }
     }
-    
-    cpu_t cpu = {.pc = 0, .sp = -1, .state = Cpu_Running, 
+
+    cpu_t cpu = {.pc = 0, .sp = -1, .csp = -1, .state = Cpu_Running,
                  .steps = 0, .stack = {0},
                  .pmem = Program};
-    
-    while (cpu.state == Cpu_Running && cpu.steps < steplimit) {
+
+    // First, handle pre-enty-point code (like exception handlers)
```

Comment:
Would you please comment on why this constraint was introduced? Why must the first instruction always be a jump?